activation/atxdb.go

```diff
@@ -366,111 +357,88 @@ func (db *ActivationDb) ActiveSetSize(epochId types.EpochId) (uint32, error) {
 	return common.BytesToUint32(val), nil
 }
 
+type atxIdAndLayer struct {
+	AtxId   types.AtxId
+	LayerId types.LayerID
+}
+
 // addAtxToEpoch adds atx to epoch epochId
 // this function is not thread safe and needs to be called under a global lock
-func (db *ActivationDb) addAtxToEpoch(epochId types.EpochId, atx types.AtxId) error {
-	ids, err := db.atxs.Get(epochId.ToBytes())
-	var atxs []types.AtxId
+func (db *ActivationDb) updateTopAtxIfNeeded(atx *types.ActivationTx) error {
+	currentTopAtx, err := db.getTopAtx()
```

Comment:
don't you need to lock this section? What if you handle new ATXs in parallel? Is it thread safe?

Reply:
The method's documentation says it's not thread safe and it should only be called under global lock. This part is not stale.

It's only called from `StoreAtx` which has a global lock from start to finish.